src/url.ts

```diff
@@ -237,8 +247,10 @@ export namespace Url {
       .map(part => part.split("="))
       .reduce(
         (memo, [param, value]) => {
+          const decodedName = decode(param);
+          const decodedValue = decode(value);
           if (param) {
-            memo[decode(param)] = decode(value);
+            memo[decodedName] ? memo[decodedName].push(decodedValue) : (memo[decodedName] = [decodedValue]);
```

Comment:
I'd prefer such variant (keep consistent variable names `param` ↔ `name`, and make it a bit easier to read):

```ts
(params, [encodedName, encodedValue]) => {
  const name = decode(encodedName);
  const value = decode(encodedValue);
  const values = memo[name] || (memo[name] = []);

  values.push(value);

  return params;
}
```

I'd drop `if (param)` to be closer with built in `URL` class:

```js
(new URL("http://example.com?=foo")).searchParams.getAll("");
// returns ["foo"]
```